packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -306,7 +303,10 @@ Dialog.slotClassNames = {
 
 /**
  * A Dialog displays important information on top of a page which usually requires user's attention, confirmation or interaction.
+ * Dialogs are purposefully interruptive, so they should be used sparingly.
+ *
  * @accessibility
  * Implements [ARIA Dialog (Modal)](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal) design pattern.
+ * Do use `trapFocus` prop to control focus trapping behavior.
```

Comment:
just a minor comment - since now we have the Best Pracitces section, I think kind of information is more visible in the Do section

Reply:
moved